tests/parsers/test.installrdf.js

```diff
@@ -22,45 +28,125 @@ describe('InstallRdfParser.getMetaData()', function() {
+        var installRdfParser = new InstallRdfParser(xmlDoc);
+        return installRdfParser._getAddonType();
+      })
+      .then((type) => {
         // Type 128 (experiments) maps to 1 PACKAGE_EXTENSION
-        assert.equal(addonMetaData.type, constants.PACKAGE_EXTENSION);
+        assert.equal(type, constants.PACKAGE_EXTENSION);
       });
   });
 
   it('should collect a notice if type is missing', () => {
     var addonValidator = new Validator({_: ['bar']});
-    var installRdfParser = new InstallRdfParser(validRDF(''),
-                                                addonValidator.collector);
-    return installRdfParser.getMetaData()
+    var rdfScanner = new RDFScanner(validRDF(''), INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc,
+                                                    addonValidator.collector);
+        return installRdfParser._getAddonType();
+      })
       .then(() => {
         var notices = addonValidator.collector.notices;
         assert.equal(notices.length, 1);
-        assert.equal(notices[0].code, messages.TYPE_MISSING.code);
+        assert.equal(notices[0].code, messages.RDF_TYPE_MISSING.code);
+      });
+  });
+});
+
+
+describe('InstallRdfParser._getName()', function() {
+
+  it('should extract a name', () => {
+    var rdf = validRDF('<em:name>my-awesome-ext</em:name>');
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc);
+        return installRdfParser._getName();
+      })
+      .then((name) => {
+        assert.equal(name, 'my-awesome-ext');
       });
   });
 
+  it('should collect a notice if name is missing', () => {
+    var addonValidator = new Validator({_: ['bar']});
+    var rdf = validRDF('<em:type>1</em:type>');
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc,
+                                                    addonValidator.collector);
+        return installRdfParser._getName();
+      })
+      .then(() => {
+        var notices = addonValidator.collector.notices;
+        assert.equal(notices.length, 1);
+        assert.equal(notices[0].code, messages.RDF_NAME_MISSING.code);
+      });
+  });
+});
+
+describe('InstallRdfParser._getGUID()', function() {
+
+  it('should extract a guid', () => {
+    var rdf = validRDF('<em:id>myid</em:id>');
+    var rdfScanner = new RDFScanner(rdf, INSTALL_RDF);
+    return rdfScanner.getContents()
+      .then((xmlDoc) => {
+        var installRdfParser = new InstallRdfParser(xmlDoc);
+        return installRdfParser._getGUID();
+      })
+      .then((guid) => {
+        assert.equal(guid, 'myid');
+      });
+  });
+
+  it('should return null for guid if not defined', () => {
```

Comment:
A note on why we do this would be handy; I could imagine stumbling in new and it taking a bit to track the "why" of this down.